src/arrays.js

```diff
@@ -8,33 +8,60 @@ const each = (elements, cb) => {
   // Iterates over a list of elements, yielding each in turn to the `cb` function.
   // This only needs to work with arrays.
   // based off http://underscorejs.org/#each
+  for (let i = 0; i < elements.length; i++) {
+    cb(elements[i], i);
+  }
 };
 
 const map = (elements, cb) => {
   // Produces a new array of values by mapping each value in list through a transformation function (iteratee).
   // Return the new array.
+  const arr = [];
+  for (let i = 0; i < elements.length; i++) {
+    arr.push(cb(elements[i]));
+  }
+  return arr;
 };
 
 const reduce = (elements, cb, memo = elements.shift()) => {
   // Combine all elements into a single value going from left to right.
   // Elements will be passed one by one into `cb`.
   // `memo` is the starting value.  If `memo` is undefined then make `elements[0]` the initial value.
+  let result = memo;
+  for (let i = 0; i < elements.length; i++) {
+    result = cb(result, elements[i]);
+  }
+  return result;
 };
 
 const find = (elements, cb) => {
+  for (let i = 0; i < elements.length; i++) {
+    if (cb(elements[i]) === true) return elements[i];
+  }
   // Look through each value in `elements` and pass each element to `cb`.
   // If `cb` returns `true` then return that element.
   // Return `undefined` if no elements pass the truth test.
 };
 
 const filter = (elements, cb) => {
+  const array = [];
+  for (let i = 0; i < elements.length; i++) {
+    if (cb(elements[i]) === true) array.push(elements[i]);
+  }
+  return array;
   // Similar to `find` but you will return an array of all elements that passed the truth test
   // Return an empty array if no elements pass the truth test
 };
 
 const flatten = (elements) => {
   // Flattens a nested array (the nesting can be to any depth).
   // Example: flatten([1, [2], [3, [[4]]]]); => [1, 2, 3, 4];
+  let array = [];
+  for (let i = 0; i < elements.length; i++) {
+    const roller = Array.isArray(elements[i]) ? flatten(elements[i]) : [elements[i]];
```

Comment:
This is good, except I don't think you want to do `[elements[i]]` at the end of your ternary. That will result in every element being wrapped inside a one-element array like `[[1], [2], [3], [4]]` instead of actually flattening the whole array.